src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -3692,6 +3702,254 @@ explicit_encryption_destroy (ee_fixture *eef)
+         ASSERT_OR_PRINT (ok, error);
+
+         ASSERT (
+            BSON_APPEND_VALUE (&to_insert, "encryptedInt", &insertPayload));
+
+         ok = mongoc_collection_insert_one (eef->encryptedColl,
+                                            &to_insert,
+                                            NULL /* opts */,
+                                            NULL /* reply */,
+                                            &error);
+         ASSERT_OR_PRINT (ok, error);
+
+         bson_value_destroy (&insertPayload);
+         bson_destroy (&to_insert);
+         mongoc_client_encryption_encrypt_opts_destroy (eopts);
+      }
+   }
+   // find operation to return both documents
+   {
+      bson_value_t findPayload;
+      mongoc_cursor_t *cursor;
+      bson_t filter = BSON_INITIALIZER;
+      const bson_t *got;
+
+      eopts = mongoc_client_encryption_encrypt_opts_new ();
+      rangeopts = mongoc_client_encryption_encrypt_range_opts_new ();
+      mongoc_client_encryption_encrypt_opts_set_query_type (
+         eopts, MONGOC_ENCRYPT_QUERY_TYPE_RANGEPREVIEW);
+      explicit_encryption_set_range_opts_int (eopts, rangeopts, eef);
+
+      bson_value_t find_doc = {0};
+      find_doc.value_type = BSON_TYPE_DOCUMENT;
+      bson_t *find = BCON_NEW ("$and",
+                               "[",
+                               "{",
+                               "encryptedInt",
+                               "{",
+                               "$gt",
+                               BCON_INT32 (5),
+                               "}",
+                               "}",
+                               "{",
+                               "encryptedInt",
+                               "{",
+                               "$lte",
+                               BCON_INT32 (30),
+                               "}",
+                               "}",
+                               "]");
+      find_doc.value.v_doc.data = (uint8_t *) bson_get_data (find);
+      find_doc.value.v_doc.data_len = find->len;
+
+      ok = mongoc_client_encryption_encrypt (
+         eef->clientEncryption, &find_doc, eopts, &findPayload, &error);
+      ASSERT_OR_PRINT (ok, error);
+
+      bson_t *findBson =
+         bson_new_from_data (findPayload.value.v_doc.data,
+                             (size_t) findPayload.value.v_doc.data_len);
+      ASSERT (!bson_empty (findBson));
+      cursor = mongoc_collection_find_with_opts (
+         eef->encryptedColl, findBson, NULL /* opts */, NULL /* read_prefs
+         */);
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_OR_PRINT (!mongoc_cursor_error (cursor, &error), error);
+      ASSERT_MATCH (got, "{ 'encryptedInt': 6}");
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_MATCH (got, "{ 'encryptedInt': 30}");
+      ASSERT (!mongoc_cursor_next (cursor, &got) &&
+              "expected 2 documents, got more than two");
+
+      bson_value_destroy (&findPayload);
```

Comment:
```suggestion
      bson_destroy (findBson);
      bson_destroy (find);
      bson_value_destroy (&findPayload);
```

To fix leaks reported by LSAN.